.github/workflows/ci.yaml

```diff
@@ -0,0 +1,245 @@
+name: CI Tasks for Ory Keto
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+
+
+jobs:
+  validate:
+    name: Run lints and checks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17
+      - run: go list -json > go.list
+      - name: Run nancy
+        uses: sonatype-nexus-community/nancy-github-action@v1.0.2
+      - name: Run prettier
+        uses: ory/ci/prettier@master
+      - name: Run prettier (docs)
+        uses: ory/ci/prettier@master
+        with:
+          dir: docs
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        env:
+          GOGC: 100
+        with:
+          args: --timeout 10m0s --issues-exit-code=0
+          skip-go-installation: true
+          skip-pkg-cache: true
+      - name: Test documentation examples
+        run: make test-docs-samples
+
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:11.8
+        env:
+          POSTGRES_DB: keto
+          POSTGRES_PASSWORD: test
+          POSTGRES_USER: test
+        ports:
+          - 5432:5432
+      mysql:
+        image: mysql:8.0
+        env:
+          MYSQL_ROOT_PASSWORD: test
+        ports:
+          - 3306:3306
+    env:
+      TEST_DATABASE_POSTGRESQL: postgres://test:test@localhost:5432/keto?sslmode=disable
+      TEST_DATABASE_MYSQL: mysql://root:test@(localhost:3306)/mysql?parseTime=true&multiStatements=true
+      TEST_DATABASE_COCKROACHDB: cockroach://root@localhost:26257/defaultdb?sslmode=disable
+    steps:
+      - run: |
+          docker create --name cockroach -p 26257:26257 \
+            cockroachdb/cockroach:v20.2.5 start-single-node --insecure
+          docker start cockroach
+        name: Start CockroachDB
```

Comment:
Why is this not a service like postgres and mysql? Just asking out of curiosity.

Reply:
GitHub, it appears, like to release half-baked features — service containers don't support overriding the default entrypoint command, or even passing args to them. And with Cockroach, we'd like to run it with the `start-single-node --insecure` args, and this isn't possible with service containers. So back to the old fashioned way of spinning up a container in the background, by hand. :P

Reply:
Haha ok, makes sense.